waku/v2/protocol/waku_swap/waku_swap.nim

```diff
@@ -207,25 +207,27 @@ proc init*(wakuSwap: WakuSwap) =
     info "received cheque", value=res.value
     wakuSwap.handleCheque(res.value)
 
-  proc credit(peerId: PeerId, n: int) {.gcsafe, closure.} =
+  proc credit(peerId: PeerId, n: int) {.closure.} =
     info "Crediting peer: ", peer=peerId, amount=n
     if wakuSwap.accounting.hasKey(peerId):
       wakuSwap.accounting[peerId] -= n
     else:
       wakuSwap.accounting[peerId] = -n
     info "Accounting state", accounting = wakuSwap.accounting[peerId]
-    waku_swap_account_state.dec(labelValues = [peerId])
+    waku_swap_account_state.set(wakuSwap.accounting[peerId], labelValues = [peerId.repr])
```

Comment:
Does this work? Could you paste a screenshot or so in PR description? Or log from Prometheus